spanner/google/cloud/spanner/database.py

```diff
@@ -154,8 +158,13 @@ def ddl_statements(self):
     def spanner_api(self):
         """Helper for session-related API calls."""
         if self._spanner_api is None:
+            base_creds = self._instance._client.credentials
+            scoped = base_creds.with_scopes((SPANNER_DATA_SCOPE,))
```

Comment:
I checked, and not in this case. the base client would've already scoped the credentials and `requires_scopes` will return false.